### ISSUE TITLE
feature request: update function for num_leapfrog_steps

### ISSUE THREAD

**Issue by soerenberg** [tensorflow/probability#266]
The Hamiltonian MCMC kernel `tfp.mcmc.HamiltonianMonteCarlo` has a very handy `step_size_update_fn`. I think it would be useful to have something similar for the number of leapfrog steps as well. Very often I found myself tweaking this parameter by hand, and if there was a possibility to set it even just to `Floor[1/step_size]` it would be very useful.

**Comment by junpenglao** [comment]
I dont think it is possible currently, as the multi-chain sampling need to synchronized across different chains, so you need to have the same number of leapfrog steps per sample. The autobatching machine @axch is working on should solve this (and NUTS).

**Comment by SiegeLordEx** [comment]
To the extent that it makes sense to share a single step size across dimensions, it makes sense to fix the product of `step_size` and `num_leapfrog_steps` the same as well. We're working on a parameter adaptation API which will definitely make this possible.

**Comment by SiegeLordEx** [comment]
As an example of this working, see the experimental NeuTra kernel: https://github.com/tensorflow/probability/blob/8fcae25982e8b5d8acfd42b8ddecf923f450ded3/experimental/neutra/neutra_kernel.py#L369-L381

If you wanted to extract that into a stable place, a possible API could look like this:

```python
kernel = tfp.mcmc.HamiltonianMonteCarlo(...)
kernel = tfp.mcmc.ConstantTrajectoryLengthStepSizeAdaptation(kernel, trajectory_length=..., target_accept_prob=..., num_adaptation_steps=...)
```

Possibly with another name, and possibly using the upcoming dual-averaging step size adaptation as the implementation.
